src/OpenSage.Game/Data/Utilities/Extensions/BinaryReaderExtensions.cs

```diff
@@ -99,7 +100,10 @@ public static string ReadFixedLengthString(this BinaryReader reader, int count)
             var result = new string(chars);
 
             // There might be garbage after the \0 character, so we can't just do TrimEnd('\0').
-            return result.Substring(0, result.IndexOf('\0'));
+            if (result.Contains('\0'))
```

Comment:
I assume fixed-length strings in `.const` files don't necessarily have a null terminator?

Reply:
Yes. Also the name ReadFixedLengthString doesn’t suggest a terminator is required. I guess you‘ve used that for W3D